### DISCUSSION TITLE
micropip - pkg that uses `scikit-learn` failing since it imports `sklearn`

### DISCUSSION THREAD

**Discussion by matthewjumpsoffbuildings** [pyodide/pyodide#2989] [Q&A] [answered]
Im trying to install a pure python wheel as follows
```js
await micropip.install('BaselineRemoval')
```
however its failing, since in the source of that package, there is a call to 
```python
from sklearn.linear_model import LinearRegression
```
this is attempting to import from the (already existing) `scikit-learn` package but it seems to be using an alias... but it fails because micropip throws this error
```
ValueError: Can't find a pure Python 3 wheel for: 'sklearn'
```
it seems that pyodide/micropip is not understanding that in the `BaselineRemoval` package, when it attempts to import `sklearn` its a reference to `scikit-learn`

i tried installing `scikit-learn` manually first using `pyodide.loadPackage` but that had no effect

**Comment by rth** [comment, marked as answer]
Thanks for the report. Opened https://github.com/scikit-learn/scikit-learn/issues/24204 for a long-term solution. 

For a short-term solution, you can explicitly install the list of dependencies (without using the `sklearn` alias) and then install the package without dependency resolution
```py
await micropip.install(['numpy', 'scipy', 'scikit-learn'])
await micropip.install('BaselineRemoval', deps=False)
```

It's also worth reporting to `BaselineRemoval` that they should not use the `sklearn` alias, and specify `scikit-learn` instead.

**Reply by lesteve** [reply]
FYI, I have made a PR in BaselineRemoval and it was merged in https://github.com/StatguyUser/BaselineRemoval/pull/9. Note that there would need to be a BaselineRemoval release on PyPI to actually fix the issue.

**Reply by rth** [reply]
Thanks, Loic!

**Comment by StatguyUser** [comment]
Hi @matthewjumpsoffbuildings, thanks for bringing it to my notice, and thank you @lesteve for the pull request. I have updated it and released it in the new version of the pypi 0.0.8 version of the package.

**Reply by lesteve** [reply]
Great, thanks!
